rocky/rocky/views/tasks.py

```diff
@@ -1,117 +1,63 @@
-        task_type = self.request.GET.get("type", self.plugin_type)
-
-        status = self.request.GET.get("scan_history_status") if self.request.GET.get("scan_history_status") else None
-
-        input_ooi = self.request.GET.get("scan_history_search") if self.request.GET.get("scan_history_search") else None
-
-        if self.request.GET.get("scan_history_from"):
-            min_created_at = datetime.strptime(self.request.GET.get("scan_history_from"), "%Y-%m-%d")
-        else:
-            min_created_at = None
-
-        if self.request.GET.get("scan_history_to"):
-            max_created_at = datetime.strptime(self.request.GET.get("scan_history_to"), "%Y-%m-%d")
-        else:
-            max_created_at = None
-
+    def get(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpResponse:
         try:
-            return client.get_lazy_task_list(
-                scheduler_id=scheduler_id,
-                task_type=task_type,
-                status=status,
-                min_created_at=min_created_at,
-                max_created_at=max_created_at,
-                input_ooi=input_ooi,
-            )
+            return super().get(request, *args, **kwargs)
+        except SchedulerError as error:
+            messages.error(request, error.message)
+        return redirect("health_beautified", organization_code=self.organization.code)
 
-        except HTTPError:
-            error_message = _("Fetching tasks failed: no connection with scheduler")
-            messages.add_message(self.request, messages.ERROR, error_message)
-            return []
+    def get_queryset(self):
+        return self.get_task_list()
 
     def post(self, request, *args, **kwargs):
-        self.handle_page_action(request.POST["action"])
-
-        return redirect(request.path)
-
-    def handle_page_action(self, action: str) -> None:
-        if action == PageActions.RESCHEDULE_TASK.value:
-            task_id = self.request.POST.get("task_id")
-            reschedule_task(self.request, self.organization.code, task_id)
+        try:
+            if self.action == self.RESCHEDULE_TASK:
+                task_id = self.request.POST.get("task_id", "")
+                self.reschedule_task(task_id)
+        except HTTPError as exc:
+            message = f"HTTP error for {exc.request.url} - {exc}"
+            messages.error(request, message)
+        except SchedulerError as error:
+            messages.error(request, error.message)
+        return super().post(request, *args, **kwargs)
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
-        try:
-            context["stats"] = client.get_task_stats(self.organization.code, self.plugin_type)
-        except SchedulerError:
-            context["stats_error"] = True
-        else:
-            context["stats_error"] = False
+        context["task_filter_form"] = self.get_task_filter_form()
```

Comment:
Evaluate lazy object
```suggestion
        context["task_list"] = list(context["task_list"]) 
        context["task_filter_form"] = self.get_task_filter_form()
```

Reply:
this did not work, because the error is thrown in `super().get_context_data` of parent `ListView` 
I made a workaround though, check it out.